src/main/java/com/aws/greengrass/shadowmanager/sync/CloudDataClient.java

```diff
@@ -283,7 +284,22 @@ private void handleUpdate(MqttMessage message) {
         String shadowName = shadowRequest.getShadowName();
         logger.atDebug().kv(LOG_THING_NAME_KEY, thingName).kv(LOG_SHADOW_NAME_KEY, shadowName)
                 .log("Received cloud update sync request");
-        syncHandler.pushLocalUpdateSyncRequest(thingName, shadowName, message.getPayload());
+        CompletableFuture
+                // Since this callback runs in context of a CRT thread, we must not block.
+                //
+                // There is a small chance of blocking when a thing/shadow lock is obtained
+                // within this call, because the lock also surrounds an IoT dataplane call
+                // in CloudUpdateSyncRequest.
+                .runAsync(() -> syncHandler.pushLocalUpdateSyncRequest(thingName, shadowName, message.getPayload()),
+                        executorService)
+                .whenComplete((unused, e) -> {
+                    if (e != null) {
+                        logger.atError().cause(e)
+                                .kv("thingName", thingName)
+                                .kv("shadowName", shadowName)
```

Comment:
nit: LOG_THING_NAME_KEY and LOG_SHADOW_NAME_KEY

Reply:
fixed